api/controller/controller.go

```diff
@@ -390,22 +390,17 @@ func (c *Client) ControllerVersion() (ControllerVersion, error) {
 	return out, err
 }
 
-// DashboardAddresses returns the address info needed to connect to the dashboard.
-func (c *Client) DashboardAddresses() ([]string, bool, error) {
-	result := params.DashboardInfo{}
-	err := c.facade.FacadeCall("DashboardAddressInfo", nil, &result)
+// DashboardConnectionInfo fetches the connection information needed for
+// connecting to the Juju Dashboard.
+func (c *Client) DashboardConnectionInfo() (params.DashboardConnectionInfo, error) {
```

Comment:
I think we generally prefer not to have params struct leak from the api packages

Reply:
yeah, this would have been carried over from way back when the gui was first added.
given we are trying to separate out a juju api, it would be good to avoid this so there's less to fix as part of that

Reply:
Oh, I looked closer, no it's new, so yeah, let's not do this.